src/main/java/com/arpnetworking/metrics/impl/TsdMetricsFactory.java

```diff
@@ -107,37 +205,66 @@ protected TsdMetricsFactory(final Builder builder) {
     public static class Builder {
 
         /**
+         * Public constructor.
+         */
+        public Builder() {
```

Comment:
This is something we've talked about before.  If you keep adding required things, you get a telescoping constructor that is really annoying.  But then using setters breaks clients silently.  There really isn't a perfect solution but we've decided to go with the getters for now.  You're right that it's a runtime break, but you do test your code, right?  ( ;